### PR TITLE
Add EIP712 feature

### PR DESCRIPTION
**Pull request by eduadiez**

### CODE REVIEW

**Review by invocamanman — APPROVED** [review]

File: src/tx-utils.js [hunk 1]
```diff
@@ -359,48 +358,52 @@ function decodeL2Tx(l2TxEncoded, nLevels){
...
+    parseChainID = parseInt(parseChainID, 16);
+
+    const domain = {
+        name: Constants.EIP712Provider,
+        version: Constants.EIP712Version,
+        chainId: parseChainID,
+        verifyingContract: parseEthAddr
+    };
+    const types = {
+        Authorise: [
```
Comment (arnaucube) [inline src/tx-utils.js:393]: `Autorise` --> `Authorize`

File: src/tx-utils.js [hunk 2]
```diff
@@ -359,48 +358,52 @@ function decodeL2Tx(l2TxEncoded, nLevels){
...
+            { name: "Provider", type: "string" },
+            { name: "Autorisation", type: "string" },
+            { name: "BJJKey", type: "bytes32" }
+        ]
+    };
+    const value = {
+        Provider: Constants.EIP712Provider,
+        Autorisation: Constants.createAccountMsg,
```
Comment (arnaucube) [inline src/tx-utils.js:401]: `Autorisation` --> `Authorization`

File: src/tx-utils.js (hunk 1 above)
Comment (arnaucube) [reply src/tx-utils.js:393]: (fixed at https://github.com/hermeznetwork/commonjs/pull/16)

File: src/tx-utils.js (hunk 2 above)
Comment (arnaucube) [reply src/tx-utils.js:401]: (fixed at https://github.com/hermeznetwork/commonjs/pull/16)

**Pull request merged**